src/System.Net.HttpListener/src/System/Net/ListenerAsyncResult.Unix.cs

```diff
@@ -0,0 +1,209 @@
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+using System.Threading;
+
+namespace System.Net
+{
+    internal class ListenerAsyncResult : IAsyncResult
+    {
+        private ManualResetEvent _handle;
+        private bool _synch;
+        private bool _completed;
+        private AsyncCallback _cb;
+        private object _state;
+        private Exception _exception;
+        private HttpListenerContext _context;
+        private object _locker = new object();
+        private ListenerAsyncResult _forward;
+        internal bool _endCalled;
+        internal bool _inGet;
+
+        public ListenerAsyncResult(AsyncCallback cb, object state)
+        {
+            _cb = cb;
+            _state = state;
+        }
+
+        internal void Complete(Exception exc)
+        {
+            if (_forward != null)
+            {
+                _forward.Complete(exc);
+                return;
+            }
+            _exception = exc;
+            if (_inGet && (exc is ObjectDisposedException))
+                _exception = new HttpListenerException((int)HttpStatusCode.InternalServerError, SR.net_listener_close);
+            lock (_locker)
+            {
+                _completed = true;
+                if (_handle != null)
+                    _handle.Set();
+
+                if (_cb != null)
+                    ThreadPool.UnsafeQueueUserWorkItem(s_invokeCB, this);
+            }
+        }
+
+        private static WaitCallback s_invokeCB = new WaitCallback(InvokeCallback);
+        private static void InvokeCallback(object o)
+        {
+            ListenerAsyncResult ares = (ListenerAsyncResult)o;
+            if (ares._forward != null)
+            {
+                InvokeCallback(ares._forward);
+                return;
+            }
+            try
+            {
+                ares._cb(ares);
+            }
+            catch
+            {
+            }
+        }
+
+        internal void Complete(HttpListenerContext context)
+        {
+            Complete(context, false);
+        }
+
+        internal void Complete(HttpListenerContext context, bool synch)
+        {
+            if (_forward != null)
+            {
+                _forward.Complete(context, synch);
+                return;
+            }
+            _synch = synch;
+            _context = context;
+            lock (_locker)
+            {
+                AuthenticationSchemes schemes = context.Listener.SelectAuthenticationScheme(context);
+                if ((schemes == AuthenticationSchemes.Basic || context.Listener.AuthenticationSchemes == AuthenticationSchemes.Negotiate) && context.Request.Headers["Authorization"] == null)
+                {
+                    context.Response.StatusCode = 401;
```

Comment:
We should replace values with the respective [HttpStatusCode](https://msdn.microsoft.com/en-us/library/system.net.httpstatuscode(v=vs.110).aspx)  equivalents.